<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="debug" default="true"/>
  <arg name="yolo_model" default="yolov8m-seg.pt"/>
  <arg name="input_topic" default="/kitti/camera_color_left/image_raw"/>
  <arg name="result_topic" default="/yolo_result"/>
  <arg name="result_image_topic" default="/yolo_image"/>
  <arg name="conf_thres" default="0.25"/>
  <arg name="iou_thres" default="0.45"/>
  <arg name="max_det" default="300"/>
  <arg name="classes" default=""/>
  <arg name="tracker" default="bytetrack.yaml"/>
  <arg name="device" default=""/>
  <arg name="result_conf" default="true"/>
  <arg name="result_line_width" default="1"/>
  <arg name="result_font_size" default="1"/>
  <arg name="result_font" default="Arial.ttf"/>
  <arg name="result_labels" default="true"/>
  <arg name="result_boxes" default="true"/>

  <node name="tracker_node" pkg="ultralytics_ros" type="tracker_node.py" output="screen">
    <param name="yolo_model" value="$(arg yolo_model)"/>
    <param name="input_topic" value="$(arg input_topic)"/>
    <param name="result_topic" value="$(arg result_topic)"/>
    <param name="result_image_topic" value="$(arg result_image_topic)"/>
    <param name="conf_thres" value="$(arg conf_thres)"/>
    <param name="iou_thres" value="$(arg iou_thres)"/>
    <param name="max_det" value="$(arg max_det)"/>
    <param name="tracker" value="$(arg tracker)"/>
    <param name="result_conf" value="$(arg result_conf)"/>
    <param name="result_line_width" value="$(arg result_line_width)"/>
    <param name="result_font_size" value="$(arg result_font_size)"/>
    <param name="result_font" value="$(arg result_font)"/>
    <param name="result_labels" value="$(arg result_labels)"/>
    <param name="result_boxes" value="$(arg result_boxes)"/>
    <rosparam param="classes" subst_value="true">$(arg classes)</rosparam>
    <rosparam param="device" subst_value="true">$(arg device)</rosparam>
  </node>

  <node if="$(arg debug)" name="image_view" pkg="image_view" type="image_view" output="screen">
    <remap from="image" to="$(arg result_image_topic)"/>
  </node>
</launch>
